### ISSUE TITLE
Binding of Isaac: Rebirth (250900)

### ISSUE THREAD

**Issue by LoreRealm** [ValveSoftware/Proton#4868]
# Compatibility Report
- Name of the game with compatibility issues:Bind of Isaac: Rebirth (NON NATIVE REPENTANCE DLC)
- Steam AppID of the game:250900
(THIS IS FOR THE REPENTANCE EXPANSION THAT ISNT LINUX NATIVE)
## System Information
- GPU: rx6800xt
- Driver/LLVM version: mesa 21>
- Kernel version: 5.12
- Link to full system information report as [Gist](https://gist.github.com/lorddickfist/3dd9fde27b9a40db2eea384036c30cc4):
- Proton version:6.3-4

## I confirm:
- [x ] that I haven't found an existing compatibility report for this game.
- [ x] that I have checked whether there are updates for my system available.

[steam-250900.log](https://github.com/ValveSoftware/Proton/files/6564017/steam-250900.log)

## Symptoms 
Controller doesn't work consistently

## Reproduction
Plug controller in, launch game try to use

**Comment by ivyl** [comment]
Can you share a few more details? What exactly do you mean by "doesn't work consistently"? Does that mean the game drops inputs or maybe the controller is not detected on each run?

What controller are you using?

Can you attach a log with `WINEDEBUG=+seh,+loaddll,+plugplay,+hid,+xinput` when the controller works and when it doesn't?

Answering those questions would help me a lot as I cannot reproduce the problem locally. I've logged about 150h playing repentance with Xbox Series X controller and I never had any issues with it. Thanks in advance!

**Comment by TheFrozenDuck** [comment]
I am not sure what the exact problem is but maybe it is the one I have. When a controller disconnects with Proton 6.3 it will not be listed in **Binding of Isaac: Repetance** anymore until the game is restarted. Can I add the variables in the Steam settings or do I need to start the application from the terminal?

**Comment by ivyl** [comment]
@TheFrozenDuck can you describe in a bit more detail what do you mean by "will no be listed"? Maybe it behaves the same on Windows? I can check that.

I have no problem with disconnecting and reconnecting the controller during gameplay. The game prompts me that the controller was lost. After reconnecting and pressing a button I am back to the game.

To get the logs you have to: right click on the game in your library -> properties -> general -> launch options:

`PROTON_LOG=1 WINEDEBUG=+seh,+loaddll,+plugplay,+hid,+xinput,+dinput %command%`

After starting the game and messing around with the controller you should see a file named `steam-250900.log` in your home directory. You can just clear "launch options" field afterwards.

**Comment by TheFrozenDuck** [comment]
> will no be listed

When going into the settings of the game you can choose an input method; be it a keyboard, xinput controller, etc. 
The controller (a Dualshock 4 in my case that is commonly listed as **xinput controller** using **proton**) is not listed in the available input options, if the controller is disconnected and reconnected to the computer. 

p.s. logs are following

**Comment by ivyl** [comment]
> a Dualshock 4 in my case that is commonly listed as xinput controller

That should not be a thing. It should be recognized as "Wireless Controller" (that's the name Sony gave it). Also hotplugging both xbox and DS4 controllers works for me with Repentance on Proton 6.3-5.

You seeing a DualShock 4 as a xinput controller can mean one of two things:

1. You don't have access to the raw HID device. Steam ships with udev rules (`/usr/lib/udev/rules.d/70-steam-input.rules` on Arch) that should tag DS4's hidraw* special file with "uaccess". It still should work just fine though.
2. You are using Steam Input (aka Steam's controller mapping) which creates a virtual xinput device. If that's the case you may be hitting [this bug](https://github.com/ValveSoftware/steam-for-linux/issues/7868). Try waiting ~60s and the device may reappear. You can also try disabling Steam Input (controller tab in game's preferences).

**Comment by malfunctioning-object** [comment]
Hello, I would also like to raise an issue for Binding of Isaac: Rebirth - Repentance DLC (Non-native).
Should I use this ticket or create a new one? It's not related to controller specifically, but overall input freezes and FPS drops.

**Comment by kisak-valve** [comment]
Hello @malfunctioning-object, in general, yes, DLC for a game should be discussed with the main game.

**Comment by malfunctioning-object** [comment]
# Compatibility Report
- Name of the game with compatibility issues: Binding of Isaac: Rebirth - Repentance DLC (Non-native Win x64 version)
- Steam AppID of the game: 250900

## System Information
- GPU: nVidia GTX 1080
- Driver/LLVM version: nvidia 470.63.01
- Kernel version: 5.13.11-1-MANJARO (also tried 5.13.arch)
- Link to full system information report as [Gist](https://gist.github.com/malfunctioning-object/31bf0474d6f5a7fb8656d9ac4484208f). (XFCE [Manjaro] / KDE Plasma [Arch], both X11)
- Proton version: 6.3.6., Experimental, (additional note - unofficial Proton GE)

## Game log
https://gist.github.com/malfunctioning-object/e4ffeadb304ed17e1a062113d94217f6

## Symptoms
Game is being processed differently somehow. It is not as stable as on Mac using CodeWeavers.
Changing rooms takes significant amount of time. Any additional sprite change causes FPS to drop. There are levels that have dropping sprites of leaking water, which also cause FPS drop. When FPS drop, input is not being processed or is just ignored/dropped. It's one of the games that should not belong to "heavy" category, yet caused the most issues.

Disabling V-Sync or lowering the quality doesn't help. Changing nvidia-settings, amount of screens or refresh rates doesn't help. 


## Reproduction
No additional specific details other than playing the game.

**Comment by malfunctioning-object** [comment]
I feel like I got hung on the "FPS drop" terminology. It feels more like a freeze/game being stuck either loading or processing stuff. I used overlay's built-in FPS monitor, even a slight drop (3-15) caused issues with game sense, movement, render.

**Comment by malfunctioning-object** [comment]
I would like to post an update to my report. 

I guess it is time to put some disclaimers that NTFS driver is not as supported as it may seem. It took a lot of tries and errors, pilling amount of issues, googling, browsing reddit, searching linux gaming discords to finally find those rare comments where people really oppose using NTFS at all. 

I had similar symptoms in other games. For example the frame issues or loading issues (initial boot could be counted in minutes, changing levels took literal ages considering the speed of the game) in BPM: Bullets Per Minute? Gone, game is blazing fast. Overall performance issues? Gone. Binding of Isaac: Repentance doesn't suffer from input lag, no loading delays, no stutters or dropped frames. Flawless gameplay. It wasn't an easy choice but at the end a worthy one.

**Comment by joseserro** [comment]
I'm having trouble with this game on the steam deck.
1) Constant crashes, sometimes when entering a new room or floor, or starting a new game
2) Freezes consistently when entering a specific level (the womb on greed), there's still a slight animation on one of the sprites and the background music still plays but the game is unresponsive to any and all inputs and doesn't load the next floor even after >20 minutes.

I've tried disabling all mods and it still happens, tried verifying game integrity, used 3 different proton versions (including Experimental, 7.0-1 and 6.3-8) and even reinstalling the game.
Nothing seems to fix this.

Is there any way to grab logs from the steam deck when it "freezes" like this?

If you need any more information, I'll gladly try to help. Thanks

Edit: managed to find a related dump and I also noticed the logs were absolutely filled with the following:

[ASSERT] - gfx/001.000_Player.anm2: No layer with Id 14

[isaacv1.7.7a.0000-20220318-022436-312-316.zip](https://github.com/ValveSoftware/Proton/files/8296082/isaacv1.7.7a.0000-20220318-022436-312-316.zip)

**Comment by kisak-valve** [comment]
The Binding of Isaac: Rebirth (250900) Afterbirth+ DLC (570660) issue on Steam Deck

Issue transferred from https://github.com/ValveSoftware/Proton/issues/5787.
@GunstarXX posted on 2022-04-23T20:56:13:

# Compatibility Report
- Name of the game with compatibility issues: The Binding of Isaac: Rebirth (Afterbirth+ DLC)
- Steam AppID of the game: 570660 (AppID of parent game, The Binding of Isaac: Rebirth (250900))

## System Information
- GPU: AMD Custom GPU 0405 (vangogh, DRM 3.45)
- Driver/LLVM version: 4.6 (Compatibility Profile) Mesa 22.0.0-devel (git-676ccacebc), LLVM 13.0.0
- Kernel version: 5.13.0-valve10.1-1-neptune-02144-g7fffaf925dfb
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/GunstarXX/90d46e15c8fe88d27b31e46731755630
- Proton version: 7.0-2

## I confirm:
- [X] that I haven't found an existing compatibility report for this game.
- [X] that I have checked whether there are updates for my system available.

## Symptoms
As long as the Afterbirth+ DLC is installed, the game will crash instantly and kick me back to Steam.

## Reproduction
Download all DLC for Binding of Isaac Rebirth up to Afterbirth+ on your Steam Deck and attempt to launch the game.

**Comment by bkosowski** [comment]
# Compatibility Report
- Name of the game with compatibility issues: The Binding of Isaac: Rebirth
- Steam AppID of the game: 250900

## System Information
- an Optimus laptop (Intel GPU + Nvidia GPU), using `prime-run` to start games and tools
- GPU: GeForce GTX 760M
- Driver/LLVM version: mesa 22.1.2-1 / nvidia 470.129.06-10
- Kernel version: 5.18.7
- Link to full system information report as [Gist](https://gist.github.com/):
  - without `prime-run`: https://gist.github.com/bkosowski/4cfe561d35a31d49c9b04ca519ec7b22
  - with `prime-run`: https://gist.github.com/bkosowski/124aca8f6db6658914047c8425609ea5
- Proton version: 7.0-3
- __GL_THREADED_OPTIMIZATION=1 LOG_PROTON=1 prime-run %command%

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

## Symptoms 

After clicking "Play", the button changes for a while (10-15 seconds), then goes to "Play" again. The game doesn't even show up.

## Reproduction

I wouldn't know how to reproduce on other systems. If happens to me every time.

[steam-250900.log](https://github.com/ValveSoftware/Proton/files/9048151/steam-250900.log)

**Comment by ivyl** [comment]
So the problem happens only when running via `prime-run`? Looks like it creates a weird mix of GL libraries:

<details>
<summary>no `prime-run` everything Intel</summary>

```
        "glx/gl" : {
          "messages" : [
            "/home/user/.local/share/Steam/ubuntu12_32/steam-runtime/amd64/usr/libexec/steam-runtime-tools-0/i386-linux-gnu-wflinfo: /home/user/.local/share/Steam/ubuntu12_32/steam-runtime/pinned_libs_32/libdbus-1.so.3: no version information available (required by /usr/lib32/libgamemode.so.0)",
            "/home/user/.local/share/Steam/ubuntu12_32/steam-runtime/amd64/usr/libexec/steam-runtime-tools-0/i386-linux-gnu-check-gl: /home/user/.local/share/Steam/ubuntu12_32/steam-runtime/pinned_libs_32/libdbus-1.so.3: no version information available (required by /usr/lib32/libgamemode.so.0)"
          ],
          "renderer" : "Mesa Intel(R) HD Graphics 4600 (HSW GT2)",
          "version" : "4.6 (Compatibility Profile) Mesa 22.1.2",
          "library-vendor" : "glvnd"
        },
        "egl_x11/gl" : {
          "messages" : [
            "/home/user/.local/share/Steam/ubuntu12_32/steam-runtime/amd64/usr/libexec/steam-runtime-tools-0/i386-linux-gnu-wflinfo: /home/user/.local/share/Steam/ubuntu12_32/steam-runtime/pinned_libs_32/libdbus-1.so.3: no version information available (required by /usr/lib32/libgamemode.so.0)"
          ],
          "renderer" : "Mesa Intel(R) HD Graphics 4600 (HSW GT2)",
          "version" : "4.6 (Compatibility Profile) Mesa 22.1.2",
          "library-vendor" : "glvnd"
        },
        "egl_x11/glesv2" : {
          "messages" : [
            "/home/user/.local/share/Steam/ubuntu12_32/steam-runtime/amd64/usr/libexec/steam-runtime-tools-0/i386-linux-gnu-wflinfo: /home/user/.local/share/Steam/ubuntu12_32/steam-runtime/pinned_libs_32/libdbus-1.so.3: no version information available (required by /usr/lib32/libgamemode.so.0)"
          ],
          "renderer" : "Mesa Intel(R) HD Graphics 4600 (HSW GT2)",
          "version" : "OpenGL ES 3.2 Mesa 22.1.2",
          "library-vendor" : "glvnd"
        }
```
</details>

<details>
<summary>`prime-run` mix of Nvidia and Intel</summary>

```
        "glx/gl" : {
          "messages" : [
            "/home/user/.local/share/Steam/ubuntu12_32/steam-runtime/amd64/usr/libexec/steam-runtime-tools-0/i386-linux-gnu-wflinfo: /home/user/.local/share/Steam/ubuntu12_32/steam-runtime/pinned_libs_32/libdbus-1.so.3: no version information available (required by /usr/lib32/libgamemode.so.0)",
            "/home/user/.local/share/Steam/ubuntu12_32/steam-runtime/amd64/usr/libexec/steam-runtime-tools-0/i386-linux-gnu-check-gl: /home/user/.local/share/Steam/ubuntu12_32/steam-runtime/pinned_libs_32/libdbus-1.so.3: no version information available (required by /usr/lib32/libgamemode.so.0)"
          ],
          "renderer" : "NVIDIA GeForce GTX 760M/PCIe/SSE2",
          "version" : "4.6.0 NVIDIA 470.129.06",
          "library-vendor" : "glvnd"
        },
        "egl_x11/gl" : {
          "messages" : [
            "/home/user/.local/share/Steam/ubuntu12_32/steam-runtime/amd64/usr/libexec/steam-runtime-tools-0/i386-linux-gnu-wflinfo: /home/user/.local/share/Steam/ubuntu12_32/steam-runtime/pinned_libs_32/libdbus-1.so.3: no version information available (required by /usr/lib32/libgamemode.so.0)"
          ],
          "renderer" : "Mesa Intel(R) HD Graphics 4600 (HSW GT2)",
          "version" : "4.6 (Compatibility Profile) Mesa 22.1.2",
          "library-vendor" : "glvnd"
        },
        "egl_x11/glesv2" : {
          "messages" : [
            "/home/user/.local/share/Steam/ubuntu12_32/steam-runtime/amd64/usr/libexec/steam-runtime-tools-0/i386-linux-gnu-wflinfo: /home/user/.local/share/Steam/ubuntu12_32/steam-runtime/pinned_libs_32/libdbus-1.so.3: no version information available (required by /usr/lib32/libgamemode.so.0)"
          ],
          "renderer" : "Mesa Intel(R) HD Graphics 4600 (HSW GT2)",
          "version" : "OpenGL ES 3.2 Mesa 22.1.2",
          "library-vendor" : "glvnd"
        }
      },
```
</details>

The game crashes somewhere in [libexpoxy](https://github.com/anholt/libepoxy) that's compiled into the game binary.

```
Backtrace:
=>0 0x00000000 (0x00e6fa34)
  1 0x009b0eb1 epoxy_gl_version+0x26b31() in isaac-ng (0x00e6fa80)
  2 0x00881db8 epoxy_handle_external_wglMakeCurrent+0x47ab28() in isaac-ng (0x00e6fee8)
  3 0x009d3576 epoxy_gl_version+0x491f6() in isaac-ng (0x00e6ff30)
  4 0x7b62ddb0 ActivateActCtx+0x23420() in kernel32 (0x00e6ff48)
  5 0x7bc5d2e7 A_SHAFinal+0x3ccc7() in ntdll (0x00e6ff5c)
  6 0x7bc5da90 A_SHAFinal+0x3d470() in ntdll (0x00e6ffec)
  ```

Can you capture logs with `prime-run` and `PROTON_LOG=+opengl,+wgl`? Is any other OpenGL game actually working for you with `prime-run`?

**Comment by bkosowski** [comment]
I cannot run the game both win `prime-run` and without it.

The logs:
`__NV_PRIME_RENDER_OFFLOAD=1 __VK_LAYER_NV_optimus="NVIDIA_only" __GLX_VENDOR_LIBRARY_NAME="nvidia" __GL_THREADED_OPTIMIZATION=1 PROTON_LOG=1 PROTON_LOG=+opengl,+wgl PROTON_USE_DXVK=1 prime-run %command%`: https://gist.github.com/bkosowski/6eeaedb8d0fae5bd299cfe1dbea48e14

`PROTON_LOG=+opengl,+wgl %command%`: https://gist.github.com/bkosowski/4fb4ab8ea6d07a4a8de4d1cd2a5e63cc

I can run:
Enter the Gungeon (OpenGL, Linux native)
Mark of the Ninja: Remastered (OpenGL, Linux native)
Never Alone (Kisima Ingitchuna) (OpenGL, Linux native)

I cannot run:
Brothers — A Tale of Two Sons (DirectX 9)
Age of Empires 2: Definitive Edition (DirectX 11)
Spelunky (DirectX 9)

So it looks like I cannot run DirectX games :thinking:

**Comment by bkosowski** [comment]
[SOLVED]

_(My solution, of course, applies only to my problem, not to the entire thread.)_

I removed these two files:
/etc/X11/xorg.conf.d/20-intel.conf
/etc/X11/10-nvidia.conf

The contents of which were:

/etc/X11/xorg.conf.d/20-intel.conf:
```
Section "Device"
  Identifier "Intel Graphics"
  Driver "intel"
  Option "TearFree" "true"
EndSection
```

/etc/X11/10-nvidia.conf:
```
Section "Files"
    ModulePath "/usr/lib/xorg/modules"
    ModulePath "/usr/lib/nvidia/xorg"
EndSection

Section "ServerLayout"
    Identifier "layout"
    Screen 0 "iGPU"
    Option "AllowNVIDIAGPUScreens"
EndSection

Section "Device"
  Identifier "iGPU"
  Driver "modesetting"
  BusID "PCI:0:2:0"
EndSection

Section "Device"
  Identifier "dGPU"
  Driver "nvidia"
  BusID "PCI:1:0:0"
EndSection

Section "Screen"
  Identifier "iGPU"
  Device "iGPU"
EndSection
```

## Additional information

I backed up and removed these two files (but I did this before and this did not fix the issue):
/usr/share/vulkan/icd.d/intel_icd.x86_64.json
/usr/share/vulkan/icd.d/intel_icd.i686.json

Before the fix, I could run:
Enter the Gungeon (OpenGL, Linux native)
Mark of the Ninja: Remastered (OpenGL, Linux native)
Never Alone (Kisima Ingitchuna) (OpenGL, Linux native)

Before the fix, I could not run:
Brothers — A Tale of Two Sons (DirectX 9)
Age of Empires 2: Definitive Edition (DirectX 11)
Spelunky (DirectX 9)

Now, I can run all games with either of these command run options:
`__GL_THREADED_OPTIMIZATION=1 __NV_PRIME_RENDER_OFFLOAD=1 __GLX_VENDOR_LIBRARY_NAME=nvidia __VK_LAYER_NV_optimus=NVIDIA_only PROTON_USE_DXVK=1 %command%`
or
`__GL_THREADED_OPTIMIZATION=1 prime-run %command%`
With the latter allowing Age of Empires 2: DE to run without crashing.

Now, `tree /etc/X11` is:
```
├── mhwd.d
│   ├── intel.conf
│   └── nvidia.conf
├── xinit
│   ├── xinitrc
│   ├── xinitrc.d
│   │   ├── 40-libcanberra-gtk-module.sh
│   │   ├── 50-systemd-user.sh
│   │   ├── 80xapp-gtk3-module.sh
│   │   └── 90xbrlapi
│   └── xserverrc
└── xorg.conf.d
    └── 00-keyboard.conf
```

**Comment by spidersouris** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/4868#issuecomment-1107648935

Got the same problem on Steam Deck. The issue does not occur with Afterbirth+ though (this DLC works fine), but with Repentance (1426300). Launching the game via Desktop Mode causes the game to crash as well. [Log](https://gist.github.com/Spidersouris/9823862baf2c5a2794841bcba80cd4e5).

**Comment by kisak-valve** [comment]
The Binding of Isaac crash

Issue transferred from https://github.com/ValveSoftware/Proton/issues/6539.
@ardishco-the-great posted on 2023-02-15T21:21:36:

# Compatibility Report
- Name of the game with compatibility issues: The Binding of Isaac Repentance
- Steam AppID of the game: 250900

## System Information
- GPU: AMD RX 6900 XT
- Driver/LLVM version: mesa 22.3.4-1
- Kernel version: 6.1.11-arch1-1
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/ardishco-the-great/9ea0237432f2b7e7398a37c30af5de5c
- Proton version: 7.0-6 and Experimental

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [X] that I have checked whether there are updates for my syste



## Symptoms 
Game crashes in local coop when a player dies
[steam-250900.log](https://github.com/ValveSoftware/Proton/files/10747954/steam-250900.log)

## Reproduction

1. Launch The Binding of Isaac with 2 or more controllers
2. Start a new run
3. Have a player die 

Using a wlroots wayland compositor might be also a step to this but It seems like the Steam Deck also has this issue, I had the exact same issue on the steam deck that uses gamescope which is an X11 session, albeit nested inside a Wayland session

**Comment by kisak-valve** [comment]
The Binding of Isaac crashes whenever a X-Box 360 Controller is plugged in (USB)

Issue transferred from https://github.com/ValveSoftware/Proton/issues/6644.
@alex-c3 posted on 2023-03-24T14:50:29:

# Compatibility Report

    Name of the game with compatibility issues:Bind of Isaac: Rebirth (NON NATIVE REPENTANCE DLC)
    Steam AppID of the game:250900
    (THIS IS FOR THE REPENTANCE EXPANSION THAT ISNT LINUX NATIVE)


## System Information
Distro:Ubuntu 22.10
Kernel:5.19.0-35-generic
RAM:16 GB
GPU Driver:4.6 Mesa 22.2.5
GPU:Intel Mesa Intel UHD 630 (CFL GT2)
CPU:Intel Core i7-8850H @ 2.60GHz
Detailed System Information : https://gist.github.com/alex-c3/d62ecaed8741a5cf7a7c10f534311009

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.


https://pastebin.com/uYWL4ehn

## Symptoms 
Whenever my X-Box 360 Controller is plugged in, the game crashes.
This means that:
1. If the controller is connected before the game launches, it will start loading to the point of showing a black blank window and crash in the next ~0.5s
2. If the game is launched without the controller plugged in, the games crashes just as the controller vibrates when plugged in (e.g. has been successfully connected). 
Notes:
a. The crash is "clean" : meaning that the window closes, and Steam detects it has closed. I can re-launch the game right after without issues, but it will crash following the patterns described above.
b. This happens using ProtonGE (7.53), ProtonExperimental and Proton 7.06
c. X-Box 360 Controller is the only USB device plugged-in.
d. Other games with Proton do work with the same controller (e.g. Hades), without crashing.
e. The game works perfectly when the controller is disconnected, as far as playing the first level.

## Reproduction
1. Install Steam
2. Force the use of Proton Experimental
3. Install The Binding Of Isaac (with Repentance)
4. Plug-in a 360 Controller
5. Launch the game

**Comment by lvlanson** [comment]
Can confirm this with other gamepads (some generic usb gamepad and xbox elite)

**Comment by Vinjul1704** [comment]
The above also happens on my end on my main PC (Arch) with an xbox-like USB gamepad attached, as well as on my Steam Deck without anything plugged in.

**Comment by powpingdone** [comment]
This seems to have been fixed by a recent steam update to the game. Dunno what's that about, but hey.

**Comment by lvlanson** [comment]
Can confirm its working again

**Comment by iodream** [comment]
# Compatibility Report
- Name of the game with compatibility issues:Binding of Isaac Repentance dlc
- Steam AppID of the game:1426300
/

## System Information
- GPU: intel hd 3000
- Driver/LLVM version: Mesa 23.1.1
- Kernel version: 6.3.6-arch1-1
- Link to full system information report as Gist: [hardware-info](https://gist.github.com/iodream/198e0f4a623562c1a2158a9b11f03b59)
- Proton version: Proton Experimental

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

## Symptoms 
The game has started crashing at random places in the game. This doesn't happen in Proton 7

## Reproduction
There doesn't seem to be a predictable pattern, other than when a layout of a level is generated, you get to a certain point, do some action like place a bomb it'll explode and it will crash there. When you hit "new game" the game will no longer crash in the same spot. If you play a lot it will crash pretty consistently, i've had about 10 crashes so far.  

[steam-250900.log](https://github.com/ValveSoftware/Proton/files/11708055/steam-250900.log)

**Comment by Vinjul1704** [comment]
I've had the above happen to me as well, but only on Proton 8. Switching to Proton 7 fixed it. I have not tried experimental. Running on a Steam Deck in my case.

**Comment by TontonSancho** [comment]
Don't know if it can be related or not :
#https://github.com/UnderscoreKilburn/repentance-issues/issues/1905

I did not yet open a bug on this repo. Some advice on the best thing to do if this finding is relevant ?

**Comment by kisak-valve** [comment]
Binding of Isaac Repentance, Bombing/Making Stone bridges crashes game on Proton 8.0.3/Experimental but not proton 7.0.x

Issue transferred from https://github.com/ValveSoftware/Proton/issues/6994.
@KoboDev posted on 2023-08-04T17:10:42:

# Compatibility Report
- Binding of Isaac Repentance
- Steam AppID of the game :250900

## System Information
- GPU: AMD ATI Radeon Vega Series / Radeon Vega Mobile Series
- Video driver version: raven2, LLVM 15.0.7, DRM 3.52, 6.4.7-arch1-2
- Kernel version: 6.4.7-arch1-2
- Link to full system information report as [Gist](https://gist.github.com/KoboDev/ca3aeb6ba9b59c5a5198d8e6b1f79e1a):
- Proton version: 8.0.3 & Experimental (as of 8/4/2023)

## I confirm:
- [Y] that I haven't found an existing compatibility report for this game.
- [Y] that I have checked whether there are updates for my system available.

## Proton Log
[steam-250900.log](https://github.com/ValveSoftware/Proton/files/12262928/steam-250900.log)


## Symptoms 
Game Crashes whenever placing down an bomb near an rock that creates an bridge.

## Reproduction
1. Select Proton 8.0.3 or Experimental
2. Launch game, play until you got an good save file (This happens randomly but often.)
3. Find an spot that requires an bridge with a bomb.
4. CRASH!

**Comment by ivyl** [comment]
I've got multiple successful runs both via the alternative mother (dross, ashpit) route and the normal one, blowing rocks left and right and I for sure created at least a few bridges and I still could not repro :-(

If someone has reached a point where you can reproduce the issue can you:
1. reach the room that has the repro (AFAIU if you crash you should restart the whole floor, so that should be easy)
2. quit the game
3. backup your `rep_persistentgamedata1.dat` or similar, you can find it in `~/.steam/root/userdata/$ACCOUNT_ID/250900/remote/` where `$ACCOUNT_ID` is a set of numbers.
4. make sure that the issue repros when you start from this save
5. attach the save file and describe exactly what to do to crash

Thanks in advance!

**Comment by ghost** [comment]
@ivyl here you have, is always crashing when you defeat the floor boss, using Proton Experimental.
[files.zip](https://github.com/ValveSoftware/Proton/files/12407491/files.zip)

**Comment by nyabinary** [comment]
@Relwi is it just me or Binding of Isaac Proton Experimental Bleeding Edge is even more broken, like every proton version, Isaac is getting even more broken

**Comment by RondoRevolution** [comment]
I can't get the game to consistently crash, every time it does, reentering the game it doesn't crash again. The most consistency I get is that it crashes almost always on Ashpit/Mines, usually by pushing a button that creates tiles on the room to walk over, by exploding bombs or when spawning an Angel Deal after beating a boss. It also happened once when entering a previously beaten boss room.

If I ever get it to consistently crash I'll attach my save file here.

Another issue I'm having with the game that I don't know if is Proton related seems to be shader related I think. I used to have more than a 100 mods a couple of months ago and it worked fine. Now having as much as 30-40 mods is enough for the game to stutter briefly when a new effect comes into the screen, the less mods I have the shorter the stutter. I can get it to shorten even further by using Zink, but I doesn't go away, with no mods it plays perfectly. After an effect happens the first time, like entering a given room, spawning an enemy, exploding, etc, it doesn't happen again on the same session. Closing and reopening Isaac makes it stuttery again.

**Comment by nyabinary** [comment]
Artifacting in Binding of Isaac Repentance in Proton Experimental and Proton 8.0, the main menu screen is fine, but once you start a new game or load into a new game, a huge black box appears with graphical artifacts making it impossible to play.

**Comment by alasky17** [comment]
@Nyabinary Additionally, you are testing on the steam deck with SteamOS Preview (3.5) (noting for future reference).  We have been completely unable to reproduce the issue you are reporting.  Could you please get a log by adding "PROTON_LOG=1 %command%" to launch options.  By default, log will be located in ~/steam-250900.log.

**Comment by cYDN48** [comment]
Also seeing unpredictable crashes on Proton 8.0-4. Here's the log of one of those crashes:

[steam-250900.log](https://github.com/ValveSoftware/Proton/files/12837734/steam-250900.log)

**Comment by nyabinary** [comment]
@alasky17 
![image](https://github.com/ValveSoftware/Proton/assets/97130632/25b947dc-ace8-4d35-96ad-fe91ea7df0ce)
Proton 7.0 also has this issue, the image is from Proton-Experimental though, in Proton 7.0 the issue exists, but the issue is much much much rarer than it is on Proton 8.0 and Proton Experimental. Also, it's random on how it works, sometime its artifacts like in the screenshot, or sometimes it's a blackbox covering the whole screen, sometimes it's just a glitchy part at the top.
[steam-250900.log](https://github.com/ValveSoftware/Proton/files/12837924/steam-250900.log)
Important to note this on my computer with an Intel CPU 12th gen, I installed and used gamescope though to make it as close as I possibly can to the Steam Deck.

**Comment by gofman** [comment]
Just updated Proton Experimental ([bleeding edge] branch, available to select on Proton Experimental tool options, BETAS page) has a possible "fix" for those random crashes. Can anyone who was reproducing those confirm if that indeed looks fixed? As I could never actually reproduce the crash even once, while found a possible reason for that.

**Comment by RondoRevolution** [comment]
> Just updated Proton Experimental ([bleeding edge] branch, available to select on Proton Experimental tool options, BETAS page) has a possible "fix" for those random crashes. Can anyone who was reproducing those confirm if that indeed looks fixed? As I could never actually reproduce the crash even once, while found a possible reason for that.

I've done 1 run so far and haven't crashed with Proton Experiental - bleeding edge while going to Ashpit/Mines. I'll try 2 more runs after I wake up and report back.

**Comment by RondoRevolution** [comment]
I couldn't test yesterday, sorry for that, but I managed to do 2 other runs today going to the alt paths using Proton Experimental - bleeding edge and I have not crashed a single time, so far it has been pretty stable.

**Comment by nyabinary** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/4868#issuecomment-1751715096

Anyone could reproduce this?

**Comment by ShayBox** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/4868#issuecomment-1712721656

I'm having the same issue with mods

**Comment by kisak-valve** [comment]
The Binding of Isaac: Repentance MALLOC crash

Issue transferred from https://github.com/ValveSoftware/Proton/issues/8246.
@LyCH-OS posted on 2024-11-15T06:31:05:

Tested with the following Proton types:
- Experimental
- 5.0-10
- 5.13-6
- 6.3-8
- 7.0-6
- 8.0-5
- 9.0-3

OS: Manjaro Linux x86_64 
Kernel: 6.11.2-zen3-xanmod1-1 
Resolution: 1920x1080, 1920x1080 
DE: Plasma 6.1.5 
WM: KWin 
CPU: AMD Ryzen 7 5800X (16) @ 4.851GHz 
GPU: AMD ATI Radeon RX 6900 XT Merc 319
Memory: 64214MiB
Mesa: 24.3.0_devel.196104.b339c525f44.d41d8cd-1

Crash Description:
After playing for an indeterminate and random time (i.e. very first room, last room of the game, entering a secret room, bombing a rock, killing an enemy, etc), the game will crash with the following terminal output,

```
Adding process 3165165 for gameID 250900
Adding process 3165166 for gameID 250900
Adding process 3165167 for gameID 250900
Adding process 3165276 for gameID 250900
esync: up and running.
wine: RLIMIT_NICE is <= 20, unable to use setpriority safely
Adding process 3165277 for gameID 250900
Adding process 3165278 for gameID 250900
Adding process 3165279 for gameID 250900
Adding process 3165282 for gameID 250900
Adding process 3165284 for gameID 250900
Adding process 3165287 for gameID 250900
Adding process 3165295 for gameID 250900
Adding process 3165301 for gameID 250900
Adding process 3165316 for gameID 250900
Adding process 3165321 for gameID 250900
Adding process 3165330 for gameID 250900
Setting breakpad minidump AppID = 250900
Steam_SetMinidumpSteamID:  Caching Steam ID:  76561197998992394 [API loaded no]
Adding process 3165344 for gameID 250900
11/15 00:11:35 minidumps folder is set to /tmp/dumps
11/15 00:11:35 Init: Installing breakpad exception handler for appid(gameoverlayui)/version(20241112170934)/tid(3165371)
11/15 00:11:35 Init: Installing breakpad exception handler for appid(gameoverlayui)/version(1.0)/tid(3165371)
malloc(): corrupted top size
```

**Comment by omfgnuts** [comment]
The controls are all semi-inverted when running Repentance+ DLC while using every Proton version (Experimenta, Hotfix, Next) except 9.0-4. Some older Proton versions work too, but they remove the ability to play new online beta

Anyone ever seen something like this or knows how to fix the issue? If I map keyboard keys to the controller buttons - it kinda works, but it's a messy solution and I wish there was something more proper of a fix :(

**Comment by simifor** [comment]
@omfgnuts could you specify what gamepad you're using? And does forcing steam input in the game properties change anything?

**Comment by omfgnuts** [comment]
> @omfgnuts could you specify what gamepad you're using? And does forcing steam input in the game properties change anything?

Any gamepad actually, I am playing on the steam deck. And even with connected dualsense it is the same. And googling this issue I see that others have it too, and they fix it by selecting the older Proton version, which is not suitable for me. I tried everything with all the Steam Input options available, no effect.

Any other logs I can provide?

**Comment by simifor** [comment]
@omfgnuts I don't get the issue on deck with its built-in controls on steamos 3.6.22, tried proton 9 and experimental. On desktop, I also didn't have issues with an xbox gamepad, however, when switching to a dualsense I saw the issue on proton 7 and up, this can be worked around by enabling steam input for the game (though you'll lose playstation glyphs).

On deck the dualsense worked for me by default as steam input was already enabled, but disabling it in game properties resulted in the game wrongly detecting that I was holding top left on the left stick.

Are you seeing something different from me? are the deck built-in controls broken for you?

**Comment by omfgnuts** [comment]
@simifor Are you on the latest Repentance+ (specifically the recent multiplayer DLC)? Because regular Repentance works just fine, and the issue appeared only after Rep+.

I'm also on latest SteamOS 3.8 (but the issue was even on the older SteamOS 3.7 version) With Proton 9 i can play multiplayer but not the Online Daily specifically (which is the last achievement I need), and controls works just fine. When switching to Experimental or any other (even GE Proton) - issue with online daily disappears but controls are all scrambled and options button also connects a second player, without any other controllers connected, and the connected second player controls perfectly (that's crazy!)

I also just tried connecting via Remote Play and use a keyboard - and it works juuust fine, even on Experimental. It's definitely not a Steam Input/Gamepad/Game issue, but i bet it's somewhere in Proton

**Comment by simifor** [comment]
@omfgnuts I have the binding of isaac rebirth, and repentance dlc, I see no mention of multiplayer in-game

**Comment by omfgnuts** [comment]
> [@omfgnuts](https://github.com/omfgnuts) I have the binding of isaac rebirth, and repentance dlc, I see no mention of multiplayer in-game

ahh, that's probably the issue. the Repentance+ DLC is free on Steam. If you have a chance - do check if you have the same issue (my friend has exactly the same prob). Maybe you'll find a workaround (playing on a keyboard via remote play is too laggy :(   )

**Comment by simifor** [comment]
@omfgnuts yeah I can see the problem with deck's controls now. Dualsense worked fine with the game IF steam input is enabled (which would make the game display xbox prompts rather than playstation ones), this is something you can force from the game's properties as a workaround for the time being.

If you want playstation glyphs, which requires disabling steam input,  I found that the controller behaved correctly when wired to the steam deck (you can use a usb c to usb c cable if you don't want to dock it), over bluetooth it would behave incorrectly.

**Comment by omfgnuts** [comment]
> [@omfgnuts](https://github.com/omfgnuts) yeah I can see the problem with deck's controls now. Dualsense worked fine with the game IF steam input is enabled (which would make the game display xbox prompts rather than playstation ones), this is something you can force from the game's properties as a workaround for the time being.
> 
> If you want playstation glyphs, which requires disabling steam input, I found that the controller behaved correctly when wired to the steam deck (you can use a usb c to usb c cable if you don't want to dock it), over bluetooth it would behave incorrectly.

I also went into the Control Settings ingame, and it shows two virtual gamepads and a keyboard connected, when in reality nothing is connected.

DualSense wired/bluetooth produces the same effect, adds a second player that controls perfectly fine, while the main character all all weird and scrambled :( 

Edit: disabling Steam Input and connecting DualSense via cable - works! woo!

**Comment by dinckelman** [comment]
The shuffled controller layouts are still present on experimental, and any other proton10 distribution I've tried. The only solution has been to roll back to proton 9. Disabling Steaminput is not an option, especially on the Deck

**Comment by zach004** [comment]
**Same issues here.**

Compatibility Report
Name of the game with compatibility issues: The Binding of Isaac: Rebirth (Repentance+ DLC)

Steam AppID of the game: 250900

System Information
GPU: AMD APU (Steam Deck)

Video driver version: Mesa (SteamOS default, latest stable as of June 2026)

Kernel version: SteamOS default (latest stable as of June 2026)

Proton version: Proton 10, Proton 11 (issue); Proton 9.x (works)

**Symptoms**

- Controller input behaves incorrectly when using Proton 10 or newer on Steam Deck.
- ABXY buttons do not function as expected
- Back button appears non-functional
- Some inputs appear inverted or incorrectly mapped
- Controls overall are inconsistent and unreliable
- The game behaves correctly with Proton 9.x.

**Reproduction**

- Launch The Binding of Isaac: Rebirth (with Repentance+ DLC) on Steam Deck
- Force compatibility to Proton 10 or Proton 11
- Start gameplay
- Attempt to use ABXY buttons and back button
- Observe incorrect or non-functional inputs

**Comment by kisak-valve** [comment]
The Binding of Isaac Uncompatible?

Issue transferred from https://github.com/ValveSoftware/Proton/issues/9932.
@BennyXZD posted on 2026-07-01T10:53:43:

Title: The Binding of Isaac: Rebirth (250900) — consistent EXCEPTION_ACCESS_VIOLATION at startup across Proton 9.0-4, GE-Proton, and Experimental
System Information:

OS: Nobara Linux 43 (Fedora 43 base), KDE Plasma 6.6.4, Wayland
Kernel: 7.0.11-200.nobara.fc43.x86_64
GPU: NVIDIA GeForce RTX 4060, driver 595.71.05 (open kernel module)
CPU: AMD Ryzen 5 7500F

Proton versions tested (all produce identical crash):

Proton 9.0-4f
GE-Proton (latest)
Proton Experimental (11.0-20260622b)

Compatibility Report Info:
Steam AppID: 250900
Symptoms:
Game window never appears (occasionally a splash flashes for a fraction of a second). Process crashes ~within seconds of launch, every time, regardless of Proton version.
Crash signature (identical across all three Proton versions):
Unhandled exception: page fault on read access to 0x00000218 in wow64 32-bit code (0x0096195d).
EIP:0096195d  EAX:00000000 EBX:000003c0 ECX:00000000 EDX:00c007e9
ESI:00bc7a30 EDI:00000000
0x0096195d isaac-ng+0x56195d: mov 0x218(%edi), %ebx
Null-pointer dereference — EDI=0, reading offset 0x218.
Notable context:

Crash occurs on a thread that was just renamed to SDL_joystick, suggesting the fault occurs during controller/input enumeration.
Immediately prior in the log: SteamAPI_Init(): Loaded 'C:\Program Files (x86)\\Steam\\steamclient.dll' OK. — native Steamworks bridge loads successfully before the crash.

Troubleshooting already attempted (no change in outcome):

Fresh Proton prefix (wiped compatdata, verified game files)
Disabled Steam Overlay
Disabled Steam Input globally (all controller support options off in Steam settings)
Unplugged physical Xbox controller and unloaded xone kernel driver modules entirely
Forced SDL_JOYSTICK_HIDAPI=0
Confirmed all DLC (Afterbirth, Afterbirth+, Repentance) installed with valid depot sizes
Verified ~/.steam symlinks are correctly pointing to Steam install
Swapped Lua5.3.3r.dll/Lua5.3.3f.dll per a known community workaround — no change
Forcing WINEDLLOVERRIDES="steamclient.dll=b,steamclient64.dll=b" avoids the crash but causes SteamAPI_Init() to fail to load entirely, and the game then exits silently without a window (not a real fix)

Attached: steam-250900.log generated with PROTON_LOG=1

[steam-250900.log](https://github.com/user-attachments/files/29548236/steam-250900.log)

**Comment by kisak-valve** [comment]
Hello @BennyXZD, please copy the contents of Steam Runtime Diagnostics from Steam (`Steam` -> `Help` -> `Steam Runtime Diagnostics`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report. `err:wgl:X11DRV_WineGL_InitOpenglInfo  couldn't initialize OpenGL, expect problems` in your Proton log hints at a video driver install issue.

**Comment by kisak-valve** [comment]
The Binding of Isaac: Rebirth (DLC REPENTENCE +) High CPU usage  100% everywhere

Issue transferred from https://github.com/ValveSoftware/Proton/issues/10131.
@yobenbobin posted on 2026-09-08T17:07:23:

 Compatibility Report
- Name of the game with compatibility issues: The Binding of Isaac: Rebirth
- Steam AppID of the game: 250900

 System Information
- GPU: MX450
- Video driver version: nvidia 610.57.04
- Kernel version: Linux 7.2.3-ogc3.1.fc44.x86_64
- Proton version: Proton 11.0-2

 I confirm:
 [✔ ] that I haven't found an existing compatibility report for this game.
[ ✔] that I have checked whether there are updates for my system available.



 Symptoms :
 CPU usage is at 100% in the lobby at all times; in the game itself, the usage doesn't drop.
This issue occurs in the Repentance+ DLC; it doesn't happen in Repentance 



Screenshots provided and logs

[steam-250900.log](https://github.com/user-attachments/files/31967520/steam-250900.log)

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/9a98ea50-dc58-480e-b70c-94f205be9ae3" />

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/a1a5b32f-a687-4866-9b4d-0880b3162baa" />

***
@yobenbobin commented on 2026-09-08T17:08:26:

Please move my request—I don't know why it ended up in the general list.
